trino/trino.sh

```diff
@@ -0,0 +1,279 @@
+    /etc/spark/conf/spark-defaults.conf |
+    tail -n 1 |
+    sed 's/.*[[:space:]=]\+\([[:digit:]]\+\).*/\1/')
+
+  local spark_executor_overhead_mb
+  if (grep spark.yarn.executor.memoryOverhead /etc/spark/conf/spark-defaults.conf); then
+    spark_executor_overhead_mb=$(grep spark.yarn.executor.memoryOverhead \
+      /etc/spark/conf/spark-defaults.conf |
+      tail -n 1 |
+      sed 's/.*[[:space:]=]\+\([[:digit:]]\+\).*/\1/')
+  else
+    # When spark.yarn.executor.memoryOverhead couldn't be found in
+    # spark-defaults.conf, use Spark default properties:
+    # executorMemory * 0.10, with minimum of 384
+    local min_executor_overhead=384
+    spark_executor_overhead_mb=$((spark_executor_mb / 10))
+    spark_executor_overhead_mb=$((spark_executor_overhead_mb > min_executor_overhead ? spark_executor_overhead_mb : min_executor_overhead))
+  fi
+  local spark_executor_count
+  spark_executor_count=$(($(nproc) / spark_executor_cores))
+
+  # Add up overhead and allocated executor MB for container size.
+  local spark_container_mb
+  spark_container_mb=$((spark_executor_mb + spark_executor_overhead_mb))
+  TRINO_JVM_MB=$((spark_container_mb * spark_executor_count))
+  readonly TRINO_JVM_MB
+
+  # Give query.max-memory-per-node 60% of Xmx; this more-or-less assumes a
+  # single-tenant use case rather than trying to allow many concurrent queries
+  # against a shared cluster.
+  # Subtract out spark_executor_overhead_mb in both the query MB and reserved
+  # system MB as a crude approximation of other unaccounted overhead that we need
+  # to leave betweenused bytes and Xmx bytes. Rounding down by integer division
+  # here also effectively places round-down bytes in the "general" pool.
+  TRINO_QUERY_NODE_MB=$((TRINO_JVM_MB * 6 / 10 - spark_executor_overhead_mb))
+  TRINO_RESERVED_SYSTEM_MB=$((TRINO_JVM_MB * 4 / 10 - spark_executor_overhead_mb))
+  readonly TRINO_QUERY_NODE_MB
+  readonly TRINO_RESERVED_SYSTEM_MB
+}
+
+function configure_node_properties() {
+  cat >/opt/trino-server/etc/node.properties <<EOF
+node.environment=production
+node.id=$(uuidgen)
+node.data-dir=/var/trino/data
+EOF
+}
+
+function configure_hive() {
+  local metastore_uri
+  metastore_uri=$(bdconfig get_property_value \
+    --configuration_file /etc/hive/conf/hive-site.xml \
+    --name hive.metastore.uris 2>/dev/null)
+
+  cat >/opt/trino-server/etc/catalog/hive.properties <<EOF
+connector.name=hive-hadoop2
+hive.metastore.uri=${metastore_uri}
+EOF
+}
+
+function configure_connectors() {
+  cat >/opt/trino-server/etc/catalog/tpch.properties <<EOF
+connector.name=tpch
+EOF
+
+  cat >/opt/trino-server/etc/catalog/tpcds.properties <<EOF
+connector.name=tpcds
+EOF
+
+  cat >/opt/trino-server/etc/catalog/jmx.properties <<EOF
+connector.name=jmx
+EOF
+
+  cat >/opt/trino-server/etc/catalog/memory.properties <<EOF
+connector.name=memory
+EOF
+  cat >/opt/trino-server/etc/catalog/bigquery.properties <<EOF
+connector.name=bigquery
+bigquery.project-id=set-your-project-id
```

Comment:
I don't think we should rely on the user editing this file directly - what if it's a cli arg such as `bigquery-project-id` and read with `/usr/share/google/get_metadata_value attributes/bigquery-project-id`?

Reply:
great point, I have added the project id of Data proc cluster and pushed the code:
] BQ_PROJECT_ID="$(/usr/share/google/get_metadata_value ../project/project-id)"